runtimes/server-interface/agent.test.ts

```diff
@@ -0,0 +1,93 @@
+import { newAgent, Agent } from './agent'
+import * as assert from 'assert'
+
+describe('Agent Tools', () => {
+    let AGENT: Agent
+    const SOME_TOOL_SPEC = {
+        name: 'test',
+        description: 'test',
+        inputSchema: {
+            type: 'object',
+            properties: {
+                test: {
+                    type: 'string',
+                },
+            },
+            required: ['test'],
+        },
+    } as const
+    const SOME_TOOL_HANDLER = async (_: { test: string }) => true
+
+    beforeEach(() => {
+        AGENT = newAgent()
+    })
+
+    it('should start without any tools', () => {
+        assert.equal(AGENT.getTools().length, 0)
+    })
+
+    it('should allow adding tools', () => {
+        AGENT.tool(SOME_TOOL_SPEC, SOME_TOOL_HANDLER)
+        assert.equal(AGENT.getTools().length, 1)
+    })
+
+    it('should allow running tools', async () => {
+        AGENT.tool(SOME_TOOL_SPEC, SOME_TOOL_HANDLER)
+        const result = await AGENT.runTool(SOME_TOOL_SPEC.name, { test: 'test' })
+        assert.equal(result, true)
+    })
+
+    it('should throw an error if tool is not found', async () => {
+        assert.rejects(async () => {
+            await AGENT.runTool(SOME_TOOL_SPEC.name, { test: 'test' })
+        }, Error)
+    })
+
+    it('should throw if the tool input does not validate', async () => {
+        AGENT.tool(SOME_TOOL_SPEC, SOME_TOOL_HANDLER)
+        assert.rejects(async () => {
+            await AGENT.runTool(SOME_TOOL_SPEC.name, { test: 1 })
+        }, Error)
+    })
+
+    it('should execute the one correct tool if multiple are available', async () => {
```

Comment:
what does "correct tool" mean here? Maybe:
```suggestion
    it('should execute the first registered tool if multiple are available', async () => {
```
?

Reply:
generally the LLM's will decide which tool is correct.  Is that what's being tested here?

Reply:
No, this is just picking the right one by name :) updated the description.